processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -244,6 +259,9 @@ public int hashCode()
     result = 31 * result + (context != null ? context.hashCode() : 0);
     result = 31 * result + (querySegmentSpec != null ? querySegmentSpec.hashCode() : 0);
     result = 31 * result + (duration != null ? duration.hashCode() : 0);
+    if (isDescending()) {
```

Comment:
can we use the standard intellij auto-generated code here or do we need it to be hand-written?